packages/shared/utils/types.ts

```diff
@@ -69,7 +69,20 @@ export type Awaitable<T> = Promise<T> | T
 
 export type ArgumentsType<T> = T extends (...args: infer U) => any ? U : never
 
-export type PromisifyFn<T extends AnyFn> = (...args: ArgumentsType<T>) => Promise<ReturnType<T>>
+/**
+ * Compatible with versions below TypeScript 4.5 Awaited
+ */
+export type Awaited<T> =
```

Comment:
A suppose `Awaited` is builtin in TypeScript isn't it?

Reply:
`Awaited` was introduced in TypeScript 4.5, here to consider compatibility.